File: backend/app/schemas/copilot.py
"""AI Copilot chat schemas."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str = Field(min_length=1, max_length=4000)


class ChatRequest(BaseModel):
    # A bounded conversation history; the last message is the new question.
    messages: list[ChatMessage] = Field(min_length=1, max_length=30)


class ChatResponse(BaseModel):
    message: str


class CopilotStatus(BaseModel):
    configured: bool
    model: str | None = None

File: frontend/src/lib/api/copilot.ts
import { apiClient } from "@/lib/api/client";

export interface ChatMessage {
  role: "user" | "assistant";
  content: string;
}

export interface CopilotStatus {
  configured: boolean;
  model: string | null;
}

export const copilotApi = {
  status: async (): Promise<CopilotStatus> =>
    (await apiClient.get<CopilotStatus>("/copilot/status")).data,
  chat: async (messages: ChatMessage[]): Promise<string> => {
    const { data } = await apiClient.post<{ message: string }>("/copilot/chat", { messages });
    return data.message;
  },
};
